appveyor.yml

```
@@ -59,8 +60,41 @@ test_script:
 - ps: |
     Write-Host "Tests are running..."
     Write-Host $pwd
-    & .\cibuild.cmd -test -integrationTest
-    if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
+    $testErrorsTotal = 0
+    $testRunCount = $env:TEST_RUN_COUNT
+    for ($testRun=1; $testRun -le $testRunCount; $testRun++)
+    {
+        Write-Host "[INFO]: Test Run ${testRun}/${testRunCount}, ${testErrorsTotal} test(s) failed before"
+
+        & .\cibuild.cmd -integrationTest
+        & .\cibuild.cmd -test
+
+        Get-ChildItem -recurse artifacts\bin\tests\TestResult.xml `
+        | ForEach-Object {
+            $resultFilePath = "$_"
+            Get-Content $resultFilePath `
+            | Select-String -Pattern '<test-run id="[^"]*"\s+testcasecount="(\d+)"\s+result="([^"]*)"\s+total="(\d+)"\s+passed="(\d+)"\s+failed="(\d+)".*' `
+            | foreach {
+                $result = $_.Matches.Groups
+                #$result[0].Value
+                $total  = $result[1].Value
+                $passed = $result[4].Value
+                $failed = $result[5].Value
+                $notPassed = $total - $passed
+                $testErrors = $failed
+                $testName = $resultFilePath.Split('\\')[-3]
+                $artifactName = "TestResult#${testRun}.${testErrors}err.${testName}.xml"
+                Push-AppveyorArtifact $resultFilePath -FileName $artifactName
+                if ($testErrors -ne 0) {
+                    $logFile = Get-ChildItem -recurse artifacts\$testname*.log
+                    Get-Content $logFile
```

Is this to output the content of the file with failed tests? Not sure, I think it may need to be:
```suggestion
                    Get-Content $logFile | Out-Host
```

Have you tested this on AppVeyor? It has its own idiosyncrasies...

---

Yes, it is the output of failed tests. It works on AppVeyor.